src/device-registry/utils/translate.js

```diff
@@ -96,35 +101,38 @@ const translateUtil = {
       for (const quiz of quizzes) {
         const translatedQuiz = { ...quiz };
         const translation = await quizTranslations(index, targetLanguage);
-        translatedQuiz.title = translation.title;
-        translatedQuiz.description = translation.description;
-        translatedQuiz.completion_message = translation.completion_message;
+        if (translation && translation.title && translation.description && translation.completion_message && translation.questions) {
 
-        const translatedQuestions = [];
-        let questionIndex = 0;
-        for (const question of quiz.questions) {
-          const translatedQuestion = { ...question };
-          const targetQuestion = translation.questions[questionIndex];
-          translatedQuestion.title = targetQuestion.title;
-          translatedQuestion.context = targetQuestion.context;
+          translatedQuiz.title = translation.title;
+          translatedQuiz.description = translation.description;
+          translatedQuiz.completion_message = translation.completion_message;
 
-          const translatedAnswers = [];
-          let answerIndex = 0;
-          for (const answer of question.answers) {
-            const translatedAnswer = { ...answer };
-            const targetAnswer = targetQuestion.answers[answerIndex];
-            translatedAnswer.title = targetAnswer.title;
-            translatedAnswer.content = targetAnswer.content;
+          const translatedQuestions = [];
+          let questionIndex = 0;
+          for (const question of quiz.questions) {
+            const translatedQuestion = { ...question };
+            const targetQuestion = translation.questions[questionIndex];
+            translatedQuestion.title = targetQuestion.title;
+            translatedQuestion.context = targetQuestion.context;
 
-            translatedAnswers.push(translatedAnswer);
-            answerIndex++;
+            const translatedAnswers = [];
+            let answerIndex = 0;
+            for (const answer of question.answers) {
+              const translatedAnswer = { ...answer };
+              const targetAnswer = targetQuestion.answers[answerIndex];
+              translatedAnswer.title = targetAnswer.title;
```

Comment:
Hi @BenjaminSsempala , thanks a lot for the recent changes. We could also do the same refactoring for the translated answers here.